data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -0,0 +1,222 @@
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileCracUtils;
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.contingency.CsaProfileContingencyCreationContext;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.triplestore.api.PropertyBag;
+import com.powsybl.triplestore.api.PropertyBags;
+
+import java.util.*;
+
+/**
+ * @author Mohamed Ben-rejeb {@literal <mohamed.ben-rejeb at rte-france.com>}
+ */
+public class CsaProfileRemedialActionsCreator {
+    private final Crac crac;
+    private final Network network;
+    private final PropertyBags gridStateAlterationRemedialActionPropertyBags;
+
+    private final PropertyBags topologyActionsPropertyBags;
+
+    private final PropertyBags contingencyWithRemedialActionsPropertyBags;
+
+    private final CsaProfileCracCreationContext cracCreationContext;
+    Set<CsaProfileRemedialActionCreationContext> csaProfileRemedialActionCreationContexts = new HashSet<>();
+
+    public CsaProfileRemedialActionsCreator(Crac crac, Network network, CsaProfileCracCreationContext cracCreationContext, PropertyBags gridStateAlterationRemedialActionPropertyBags, PropertyBags topologyActionsPropertyBags, PropertyBags contingencyWithRemedialActionsPropertyBags) {
+        this.crac = crac;
+        this.network = network;
+        this.gridStateAlterationRemedialActionPropertyBags = gridStateAlterationRemedialActionPropertyBags;
+        this.topologyActionsPropertyBags = topologyActionsPropertyBags;
+        this.contingencyWithRemedialActionsPropertyBags = contingencyWithRemedialActionsPropertyBags;
+        this.cracCreationContext = cracCreationContext;
+        createAndAddRemedialActions();
+    }
+
+    private void createAndAddRemedialActions() {
+        Map<String, Set<PropertyBag>> linkedTopologyActions = CsaProfileCracUtils.getMappedPropertyBagsSet(topologyActionsPropertyBags, CsaProfileConstants.GRID_STATE_ALTERATION_REMEDIAL_ACTION);
+        Map<String, Set<PropertyBag>> linkedContingencyWithRAs = CsaProfileCracUtils.getMappedPropertyBagsSet(contingencyWithRemedialActionsPropertyBags, CsaProfileConstants.GRID_STATE_ALTERATION_REMEDIAL_ACTION);
+
+        for (PropertyBag parentRemedialActionPropertyBag : gridStateAlterationRemedialActionPropertyBags) {
+            String remedialActionId = parentRemedialActionPropertyBag.get(CsaProfileConstants.MRID);
+
+            try {
+                if (!linkedTopologyActions.containsKey(remedialActionId)) {
+                    csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "Remedial Action: " + remedialActionId + " will not be imported because there is no topology actions linked to that RA"));
+                    continue;
+                }
+
+                checkRemedialActionCanBeImported(parentRemedialActionPropertyBag);
```

Comment:
to skip the RA if there's a problem and continue to the next RA, surround this with try/catch
```suggestion
try {
                checkRemedialActionCanBeImported(parentRemedialActionPropertyBag);
                } catch (FaraImportException e) {             csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, e.getImportStatus(), e.getMessage()));
                    continue;
                }
```